bindings/java/src/main/java/org/openucx/jucx/ucp/UcpContext.java

```diff
@@ -61,12 +62,62 @@ public UcpMemory registerMemory(ByteBuffer buf) {
         if (!buf.isDirect()) {
             throw new UcxException("Registered buffer must be direct.");
         }
-        return registerMemoryNative(getNativeId(), buf);
+        UcpMemory result = registerMemoryNative(getNativeId(), UcxUtils.getAddress(buf),
+            buf.remaining(), 0L);
+        // To keep reference to this buffer, so it won't be cleaned by GC.
+        result.setData(buf);
+        return result;
+    }
+
+    /**
+     * Associates memory allocated/mapped region with the network resources.
+     * The network stack associated with an application context
+     * can typically send and receive data from the mapped memory without
+     * CPU intervention; some devices and associated network stacks
+     * require the memory to be mapped to send and receive data.
+     * Complete the mapping faster, possibly by not populating the pages in the mapping up-front,
+     * and mapping them later when they are accessed by communication routines.
+     */
+    public UcpMemory registerMemoryNonBlocking(ByteBuffer buf) {
+        if (!buf.isDirect()) {
+            throw new UcxException("Registered buffer must be direct.");
+        }
+        UcpMemory result = registerMemoryNative(getNativeId(), UcxUtils.getAddress(buf),
+            buf.remaining(), UcpConstants.UCP_MEM_MAP_NONBLOCK);
+        result.setData(buf);
+        return result;
+    }
+
+    public UcpMemory registerMemory(long address, long length) {
+        return registerMemoryNative(getNativeId(), address, length, 0L);
```

Comment:
should `0L` be `UCP_MEM_MAP_NONBLOCK`?

Reply:
There's 2 versions of API: `registerMemory` default and `registerMemoryNonBlocking` where it uses `UCP_MEM_MAP_NONBLOCK` flag.